fe/fe-core/src/main/java/org/apache/doris/planner/HashDistributionPruner.java

```diff
@@ -79,10 +80,13 @@ public Collection<Long> prune(int columnId, PartitionKey hashKey, int complex) {
             return Lists.newArrayList(bucketsList.get((int) ((hashValue & 0xffffffff) % hashMod)));
         }
         Column keyColumn = distributionColumns.get(columnId);
-        String columnName = isBaseIndexSelected ? keyColumn.getName()
-                : org.apache.doris.nereids.rules.rewrite.mv.AbstractSelectMaterializedIndexRule
+        String columnName = keyColumn.getName();
+        if (!isBaseIndexSelected && ConnectContext.get().getSessionVariable().isEnableNereidsPlanner()) {
```

Comment:
use isEnableNereidsPlanner is not a good way, because we could fallback to legacy planner, even if isEnableNereidsPlanner is true. maybe u could use `org.apache.doris.qe.QueryState#isNereids`

Reply:
fallback to legacy planner also OK, as this pr fix the issue in legacy planner